### PR TITLE
Conditionally build WildFly adapters for our testsuite

### LINKED ISSUE #20077: Conditionally build WildFly adapters for our testsuite
### Description

Current distributions of WildFly adapters are never used in our testsuite, only ones fetched from the latest nightly build. We could also prepare a solution for our GitHub Actions CI to not build these adapters for each run, but only if there are some changes. If not, you can use the built adapters from the nightly build.

### PR DESCRIPTION
**Pull request by mabartos**
Fixes #20077

Revert once https://github.com/keycloak/keycloak/issues/19299 is solved

### CODE REVIEW

**Comment by mabartos** [comment]
@vmuzikar Could you please check it?

With this approach, we save more than 5 minutes for `Keycloak CI / Build` as we don't need to build the WildFly adapters always. 

These adapters are only built if in a sent PR, there's some change made for OIDC or SAML WildFly adapter. Or in case there were some changes made in different PR that day and merged into the main. 

Otherwise, WildFly adapters are used from the latest nightly build.

-------------------------------
 The current approach works as expected. The profile is properly specified when there are some changes. See: https://github.com/keycloak/keycloak/pull/20079
 
Build time

- this PR (no need to build WF adapters): 14m 57s
- test PR #20079 (explicitly build WF adapters):  20m 36s

**Comment by mabartos** [comment]
I need to analyze a particular issue with the diffs.

**Review by vmuzikar — DISMISSED** [review]
LGTM, thanks @mabartos.

Before merging, I think we should also get an approve from someone from the core team.
CC @mposolda @pedroigor

File: .github/actions/build-keycloak/action.yml
Comment (mabartos) [inline .github/actions/build-keycloak/action.yml:68]: @stianst Could you please check it, as you've been working on the GHAs? Thank you

**Comment by mabartos** [comment]
Resolved conflict caused by the change in https://github.com/keycloak/keycloak/commit/2a18bcb246adcbbf8c5c77d89d1fffaa108df20c#diff-e05052530aff0c05c1adaf736274bb7d4a18b4992ece8b17f880b58b37260a5aL42-L45.

**Comment by mabartos** [comment]
With @miquelsi, we've encountered an issue related to specifying Maven profiles for enabling various modules. It's also the Maven convention that you should specify only direct children submodules. I think it'd be better to build a whole distribution package when there are some changes for the WildFly adapters in order to prevent some other issues related to this. And as this is only a temporary solution, I don't see any serious problems with going this way. 

I'll check what's the difference in the build time, but it might not be a big difference. 

@vmuzikar Are you ok with that?

**Review by vmuzikar — DISMISSED** [review]
@mabartos Works for me, thanks.

All we need now is a core team review. :)

**Comment by mabartos** [comment]
@vmuzikar One more thing changed; added dependency for SAML adapter.

**Review by vmuzikar — APPROVED** [review]

**Review by miquelsi — APPROVED** [review]
Verified it works in internal build pipelines. Thanks @mabartos!

**Comment by mabartos** [comment]
@vmuzikar Do we really need approval from the Core team? These changes are related only to the testsuite and our GHA CI. Seems guys from the core team don't have the capacity to review it right now.

**Comment by vmuzikar** [comment]
We'll probably need an admin override (maybe from @mposolda) in any case to merge this due to the failed UI tests, I'm afraid.

**Review by mposolda — APPROVED** [review]

**Pull request merged**
